src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/CourseController.java

```diff
@@ -90,9 +106,28 @@ public String add(ScheduleItem scheduleItem,
         primary.setLocation(lecture_location);
         primary.setQuarter(lecture_quarter);
         logger.info("primary = " + primary); 
+      
+        //here we find and delete the courses if they already exist.
+        List<ScheduleItem> myClasses = scheduleItemRepository.findByUid(scheduleItem.getUid());
+        myClasses = deleteByClassname(myClasses, scheduleItem.getClassname());
+
         scheduleItemRepository.save(primary);
+        myClasses.add(primary);
+        scheduleItemRepository.save(scheduleItem);
+        myClasses.add(scheduleItem);
 
-        model.addAttribute("myclasses", scheduleItemRepository.findByUid(scheduleItem.getUid()));
+        model.addAttribute("myclasses", myClasses);
+        return "courseschedule/index";
+    }
+
+    @PostMapping("/courseschedule/delete")
+    public String delete(ScheduleItem scheduleItem, Model model){
+        logger.info("ScheduleItem to delete: " + scheduleItem.getUid());
+        List<ScheduleItem> myClasses = scheduleItemRepository.findByUid(scheduleItem.getUid());
+        List<ScheduleItem> remainingClasses = deleteByClassname(myClasses, scheduleItem.getClassname());
```

Comment:
If deleteByClassname simply modified the list passed to it, then you wouldn't need two separate lists. Is there a reason this approach wasn't taken?